flink-core/src/main/java/org/apache/flink/metrics/groups/AbstractMetricGroup.java

```diff
@@ -100,10 +101,29 @@ public AbstractMetricGroup(MetricRegistry registry, String[] scope) {
 	 * @return fully qualified metric name
      */
 	public String getMetricIdentifier(String metricName) {
+		return getMetricIdentifier(metricName, null);
+	}
+
+	/**
+	 * Returns the fully qualified metric name, for example
+	 * {@code "host-7.taskmanager-2.window_word_count.my-mapper.metricName"}
+	 *
+	 * @param metricName metric name
+	 * @param filter character filter which is applied to the fully qualified metric name
+	 * @return fully qualified metric name
+	 */
+	public String getMetricIdentifier(String metricName, CharacterFilter filter) {
 		if (scopeString == null) {
-			scopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
+			if (filter != null) {
+				scopeString = ScopeFormat.concat(registry.getDelimiter(), scopeComponents);
```

Comment:
if no filter is given we will now never assign to scopeString, breaking all names. We should assume that no filtering is required.